.github/service/model.py

```diff
@@ -0,0 +1,144 @@
+    # INSTANCE METHODS
+    ##################################################
+
+    def __repr__(self):
+        return "<Recommendation %d %d %d>" % (self.idA, self.idB, self.typeid)
+
+    def create(self):
+        """ 
+        Creates a recommendation pair to the database
+        """ 
+        logger.info("Creating %s", self.name)
+        self.id = None  # id must be none to generate next primary key
+        db.session.add(self)
+        db.session.commit()
+
+    def save(self):
+        """
+        Updates a recommendation to the database
+        """
+        logger.info("Saving %s", self.idA)
+        db.session.commit()
+    
+    def delete(self):
+        """ Removes all recommendation from the data store by using product id"""
+        logger.info("Deleting %s", self.idA)
+        db.session.delete(self)
+        db.session.commit()
+
+    def serialize(self):
+        """ Serializes a recommendation into a dictionary """
+        return {
+            "productA-id": self.idA,
+            "productB-id": self.idB,
+            "type-id": self.typeid
+        }
+
+    def deserialize(self, data):
+        """
+        Deserializes a recommendation from a dictionary
+        Args:
+            data (dict): A dictionary containing the resource data
+        """
+        try:
+            self.idA = data["productA-id"]
+        except KeyError as error:
+            raise DataValidationError("Invalid recommendation: missing " + error.args[0])
+        except TypeError as error:
+            raise DataValidationError(
+                "Invalid recommendation: body of request contained" "bad or no data"
+            )
+        return self
+
+    ##################################################
+    # CLASS METHODS
+    ##################################################
+
+    @classmethod
+    def init_db(cls, app):
+        """ Initializes the database session """
+        logger.info("Initializing database")
+        cls.app = app
+        # This is where we initialize SQLAlchemy from the Flask app
+        db.init_app(app)
+        app.app_context().push()
+        db.create_all()  # make our sqlalchemy tables
+
+    @classmethod
+    def all(cls):
+        """ Returns all of the recommendations in the database """
+        logger.info("Processing all recommendations")
+        return cls.query.all()
+
+    @classmethod
+    def find(cls, by_id):
+        """ Finds a recommendation by it's ID """
+        logger.info("Processing lookup for id %s ...", by_id)
+        return cls.query.get(by_id)
+
+    @classmethod
+    def find_or_404(cls, by_id):
+        """ Find a recommendation by it's id """
+        logger.info("Processing lookup or 404 for id %s ...", by_id)
+        return cls.query.get_or_404(by_id)
+
+    @classmethod
+    def find_by_name(cls, name):
+        """ Returns all recommendations with the given name
+        Args:
+            name (string): the name of the recommendations you want to match
```

Comment:
Do we have a "name" field for a Recommendation?

Reply:
No, we don't. So that's why I comment it in the first pull request. Should I delete this part first? And we can add it later in the future if it's needed?

Reply:
Yes that is what I meant in my initial comment, sorry if it wasn't clear. We can later add a Github Issue in the future.